src/NPRKTrees.jl

```diff
@@ -0,0 +1,173 @@
+		for tree in RootedTreeIterator(order)
+			for c in colors
+				colorseq = collect(c)
+				push!(toReturn, rootedtree(tree.level_sequence,colorseq))
+			end
+		end
+	end
+	# remove isomorphic trees
+	toReturn = unique(toReturn,dims=1)
+	return toReturn
+end;
+
+# Generates all non-isomorphic NPRK trees with num_partitions (i.e., edge colors) of a given order
+#   equivalent to all ARK trees with num_partition colors of that order, with root node coloring fixed = 0
+# 	returned as a Vector of ColoredRootedTree's
+function NPRKTreeIterator(num_partitions::Int64, order::Int64)::Vector{<:ColoredRootedTree}
+	toReturn = ColoredRootedTree[]
+	
+	if num_partitions < 1
+		throw(DomainError(num_partitions,"num_partitions argument must be integer >= 1"))
+	elseif order < 1
+		throw(DomainError(order,"order argument must be integer >= 1"))
+	elseif order == 1
+		# if order is 1, generate trivial tree with fixed root color 0
+		trivialtree = rootedtree([1],[0])
+		push!(toReturn,trivialtree)
+	else
+		# else order > 1:
+		# Generate all possible color sequences with colorings 0,1,...,partitions-1
+		# 	of length order-1 (minus one because the root coloring is fixed to 0)
+		colors = reverse.(Iterators.product(fill(0:num_partitions-1,order-1)...))[:]
+	
+		# Loop over all uncolored rooted trees and append the above color sequences
+		# 	to define colored rooted trees with fixed root color 0
+		# Note that this will generate duplicate trees that are isomorphic
+		for tree in RootedTreeIterator(order)
+			for c in colors
+				colorseq = collect(c)
+				pushfirst!(colorseq,0) # push 0 to start of color sequence
+				push!(toReturn, rootedtree(tree.level_sequence,colorseq))
+			end
+		end
+	end
+	# remove isomorphic trees
+	toReturn = unique(toReturn,dims=1)
+
+	return toReturn
+end;
+
+# Determines whether a ColoredRootedTree is multicolored:
+# 	argument root_color_fixed = true for NPRK trees, i.e., checks if edges of the NPRK tree are multicolored
+# 	argument root_color_fixed = false for ARK trees, i.e., checks if nodes of the NPRK tree are multicolored
+# 		corresponds to additive coupling conditions
+function isMultiColored(colored_tree::ColoredRootedTree, root_color_fixed::Bool)::Bool
+	if root_color_fixed
+		colorseq = copy(colored_tree.color_sequence)
+		popfirst!(colorseq)
+		return ~allequal(colorseq)
+	else
+		return ~allequal(colored_tree.color_sequence)
+	end
+end
+
+# For a given tree's level sequence and a given node in that level sequence specified by position nodeindex in the level sequence
+# 	return the index in the level sequence of the parent node
+#   note that if nodeindex == 1, i.e., the node specified is the root node, this returns 0 since the root has no parent
+function GetParentIndex(levelseq::Vector, nodeindex::Int64)::Int64
+	toReturn = 0
+	for i in reverse(1:length(levelseq))
+		if levelseq[i] == levelseq[nodeindex]-1
+			toReturn = i
+			break
+		end
+	end
+	return toReturn
+end
```

Comment:
```suggestion
function get_parent_index(levelseq::AbstractVector, nodeindex::Integer)
	to_return = 0
	for i in reverse(eachindex(levelseq))
		if levelseq[i] == levelseq[nodeindex] - 1
			to_return = i
			break
		end
	end
	return to_return
end
```

See the comments on the style above. In general, Julia does not require strict type constraints in function arguments; see, e.g., https://docs.julialang.org/en/v1/manual/style-guide/#Avoid-writing-overly-specific-types